ReviewChecklist.md

```diff
@@ -0,0 +1,167 @@
+# PR Review Checklist
+The Solidity compiler is a critical piece of infrastructure in the Ethereum ecosystem.
+For this reason, our review process is quite strict and all PRs have to fulfill certain quality
+expectations and guidelines.
+The list below is meant to reduce the workload on the core team by helping contributors self-identify
+and solve common issues before they are pointed out in the review.
+It is also meant to serve as a final checklist for reviewers to go through before approving a PR.
+
+## Before You Submit a PR
+- [ ] **Is the issue ready to be worked on?**
+    - If the issue does not have a desirability label (`nice to have`, `should have`,
+        `must have eventually`, `must have`, `roadmap`) we have not yet decided whether to implement it.
+    - If the issue has the `needs design` label, we have not yet decided how it should be implemented.
+    - `good first issue candidate` means that the issue will potentially be a `good first issue`
+        eventually but at the moment it is not yet ready to be worked on.
+- [ ] **Is this a breaking change?** Breaking changes should be based on the `breaking` branch rather than on the `develop` branch.
+- [ ] **Does the PR actually address the issue?**
+    - [ ] Mention the issue number in the PR description.
+        If the PR solves it completely, use the `Fixes #<issue number>` form so that Github will close the issue automatically.
+    - [ ] Do not include the issue number in the PR title, branch name or commit description.
+- [ ] When submitting a PR from a fork **create a branch and give it a descriptive name.**
+    E.g. `fix-array-abi-encoding-bug`.
+    Do not submit PRs directly from the `develop` branch of your fork since it makes rebasing and fetching new changes harder.
+- [ ] **Does the PR depend on other PRs?**
+    - [ ] If the PR has dependencies, mention them in bold in the description.
```

Comment:
Was boldness used much for this before?

Reply:
I used bold mostly because the explanations get in the way a bit and the list turned out to be more verbose that I'd like it to. So I bolded the main bit of each point so that if you went through the list once, next time you can just focus on those bits and mostly ignore the rest.

Any alternative ideas?

Reply:
I referred to
> If the PR has dependencies, mention them **in bold in the description**

:)

Reply:
Haha, you got me.

I don't think we have such a rule but I think it's a good idea anyway to make it more visible. I used to make the bit about dependencies bold in my own PRs until we added `has dependencies` label. Now I usually don't but a contributor submitting a PR can't add a label so bolding it will help us spot that.